telemetry/telemetry.go

```diff
@@ -112,7 +114,11 @@ func (reportMgr *ReportManager) SendReport(tb *TelemetryBuffer) error {
 		report, err = reportMgr.ReportToBytes()
 		if err == nil {
 			if _, err = tb.Write(report); err != nil {
-				log.Printf("telemetry write failed:%v", err)
+				if reportMgr.Logger != nil {
```

Comment:
If this is the only place report manager is logging using zap logger, should we use `tb.logger` here as well ?
In that case we remove the `Logger` from `ReportManager` struct ?